Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -28,15 +29,35 @@ export class azureclitask {
             var connectedService: string = tl.getInput("connectedServiceNameARM", true);
             this.loginAzureRM(connectedService);
 
+            let errLinesCount: number = 0;
+            let aggregatedErrorLines: string[] = [];
+            tool.on('errline', (errorLine: string) => {
+                if (errLinesCount < 10) {
+                    aggregatedErrorLines.push(errorLine);
+                }
+                errLinesCount++;
+            });
+
             var addSpnToEnvironment: boolean = tl.getBoolInput("addSpnToEnvironment", false);
             if (!!addSpnToEnvironment && tl.getEndpointAuthorizationScheme(connectedService, true) == "ServicePrincipal") {
-                await tool.exec({
-                    failOnStdErr: failOnStdErr,
+                exitCode = await tool.exec({
+                    failOnStdErr: false,
+                    ignoreReturnCode: true,
                     env: { ...process.env, ...{ servicePrincipalId: this.servicePrincipalId, servicePrincipalKey: this.servicePrincipalKey, tenantId: this.tenantId } }
                 });
             }
             else {
-                await tool.exec({ failOnStdErr: failOnStdErr });
+                exitCode = await tool.exec({
+                    failOnStdErr: false,
+                    ignoreReturnCode: true
+                 });
+            }
+
+
+            if (failOnStdErr) {
```

Comment:
should we do this irrespective of whether failOnStdErr is true or false? If exit code is non zero, even then we should show the error lines to user.

Reply:
Okay, I think above line is not easy to understand. What I meant was whether we should do this `if (failOnStdErr || (exitCode != 0)) {`

Reply:
yes that can be added, but the aggregatedErrorLines will be empty if there is no error i.e. if exitCode = 0. So it will no print anything.